app/models/teacher_set.rb

```diff
@@ -665,4 +667,32 @@ def as_json(opts={})
   end
 =end
 
+  # Recieve JSON related to a teacher_set.
+  # For each ISBN, ensure there is an associated book.
+  # Disassociate books that are no longer in the teacher set.
+  def update_included_book_list(teacher_set_record)
```

Comment:
Method looks great. You can also break the blocks of code inside the method to 2-3 other methods. Something to think about.

Reply:
Thinking out loud here; I welcome more discussion on it...
I cleaned up the method so that it's more readable now.  Because our team's norm is to add comments to methods, I think there's less of a need to break larger methods into smaller ones.  The primary advantage of doing so is to increase readability (many people create readable method names to serve as code comments), however we've got long comments which are more descriptive than those short method names would be.  

These would be private methods so they wouldn't be tested, so there isn't much of a reliability advantage, and it would be more lines of code in total with a chance that a variable isn't available in the private method's scope if it isn't passed in, so I recommend that we stick with what we've got.